src/instructlab/eval/unitxt.py

```diff
@@ -0,0 +1,127 @@
+"""
+Unitxt - Flexible, Shareable and Reusable Data Preparation and Evaluation for Generative AI
+https://github.com/IBM/unitxt
+https://arxiv.org/abs/2401.14019
+"""
+
+# Standard
+from uuid import uuid4
+import os
+import shutil
+
+# Third Party
+from lm_eval.tasks.unitxt import task
+import yaml
+
+# First Party
+from instructlab.eval.mmlu import MMLUBranchEvaluator
+
+# Local
+from .logger_config import setup_logger
+
+logger = setup_logger(__name__)
+
+TEMP_DIR_PREFIX = "unitxt_temp"
+
+
+class UnitxtEvaluator(MMLUBranchEvaluator):
+    """
+    An evaluator class, running Unitxt evaluation
+
+    Attributes:
+        model_path      absolute path to or name of a huggingface model
+        unitxt_recipe   unitxt recipe (see unitxt.ai for more information)
+                        A Recipe holds a complete specification of a unitxt pipeline
+                        Example: card=cards.wnli,template=templates.classification.multi_class.relation.default,max_train_instances=5,loader_limit=20,num_demos=3,demos_pool_size=10
+
+    """
+
+    name = "unitxt"
+
+    def __init__(
+        self,
+        model_path,
+        unitxt_recipe: str,
+    ):
+        unitxt_task = self.assign_task_name()
+        tasks_dir = self.assign_tasks_dir(unitxt_task)
```

Comment:
Is this using a local directory?  If so, it needs to be built off a param like output_dir with mt_bench.

Reply:
this is a temporary directory, deleted at the end of the evaluation process. Would you prefer the user specified an output dir? It does not contain anything of use for the user, just the files required for lm eval to run unitxt.

Reply:
I think it would make sense to use the output_dir so it doesn't confuse the user in a local directory.  Also, since you do want to remove at the end, the create/remove logic should probably be in a try/finally block.

Reply:
so a user would specify an output dir but will find it doesn't exist at the end of the run?
If the user specifies it, I guess I will not delete it, right?

Reply:
Would it make sense to put all this into a [memory filesystem](https://docs.pyfilesystem.org/en/latest/reference/memoryfs.html)?  In general, it is best to avoid unnecessary disk writes, especially for something that's likely to run in a cloud service where it may or may not have write permissions on some sort of disk.

Reply:
memory filesystem does not seem to work well, as directory is later accessed also by lm-eval inside the mmlu class and I don't want to start passing this filesystem around (unless owners support such an overall change)